tier1/keyvalues3.cpp

```diff
@@ -533,62 +524,72 @@ void KeyValues3::SetColor( const Color &color )
 
 int KeyValues3::GetArrayElementCount() const
 {
-	if ( GetType() != KV3_TYPE_ARRAY )
+	const CKeyValues3Array *pArray = GetArray();
+
+	if ( !pArray )
 		return 0;
 
-	if ( GetTypeEx() == KV3_TYPEEX_ARRAY )
-		return m_Data.m_pArray->Count();
-	else
-		return m_nNumArrayElements;
+	return IsArray() ? pArray->Count() : m_nNumArrayElements;
```

Comment:
In which cases does m_nNumArrayElements hit in here? Since GetArray already does IsArray check, so it would always be true in here.

Reply:
No, `IsArray`/`IsTable` hasn't ex check